File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.cpp
class Solution {
public:
    string reverseVowels(string s) {
        unordered_set<char> vowel={'a','e','i','o','u','A','E','I','O','U'};
        int i=0;
        int j=s.length()-1;
        while(i<j){
         if(vowel.find(s[i])==vowel.end()){
             i++;
         }else if(vowel.find(s[j])==vowel.end()){
             j--;
         }else{
             swap(s[i],s[j]);
             i++;
             j--;
         }
        }
        return s;
    }
};

File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.java
class Solution {
    public String reverseVowels(String s) {

        Set<Character> set=new HashSet<>(Arrays.asList('a','e','i','o','u','A','E','I','O','U'));

        char[] chars=s.toCharArray();

        int start=0;
        int end=s.length()-1;
        while(start<end){
            if(set.contains(chars[start]) && set.contains(chars[end])){
                char temp=chars[start];
                chars[start]=chars[end];
                chars[end]=temp;
                start++;
                end--;
            }else if(!set.contains(chars[start])){
                start++;
            }else{
                end--;
            }
        }
        return new String(chars);
    }
}